tests/h/h_api/bulk_api/entry_point_test.py

```diff
@@ -74,14 +75,18 @@ def test_we_reject_bad_arguments(self, method, kwargs, executor):
     def test_to_stream(self, commands):
         handle = StringIO()
 
+        expected = deepcopy([command.raw for command in commands])
+
         BulkAPI.to_stream(handle, commands)
 
-        self._assert_nd_json_matches_commands(handle.getvalue(), commands)
+        assert self._decode_ndjson(handle.getvalue()) == expected
 
     def test_to_string(self, commands):
+        expected = deepcopy([command.raw for command in commands])
+
         nd_json = BulkAPI.to_string(commands)
 
-        self._assert_nd_json_matches_commands(nd_json, commands)
+        assert self._decode_ndjson(nd_json) == expected
```

Comment:
Due to query merging, we need to capture the expected output before we process the queries (as the post merge isn't what we send)